NGitLab.Mock.Tests/BranchMockTests.cs

```diff
@@ -0,0 +1,39 @@
+using System.Linq;
+using NGitLab.Mock.Config;
+using NUnit.Framework;
+
+namespace NGitLab.Mock.Tests
+{
+    public class BranchMockTests
+    {
+        [Test]
+        public void Test_search_branches()
+        {
+            using var server = new GitLabConfig()
+                .WithUser("user1", isDefault: true)
+                .WithProject("test-project", id: 1, addDefaultUserAsMaintainer: true, defaultBranch: "main", configure: project => project
+                    .WithCommit("Initial commit")
+                    .WithCommit("Commit for branch_1", sourceBranch: "branch_1"))
+                .BuildServer();
+
+            var client = server.CreateClient("user1");
+            var branchClient = client.GetRepository(1).Branches;
+
+            var branches = branchClient.Search("main").ToList();
+            Assert.IsTrue(branches.Any(branch => string.Equals(branch.Name, "main", System.StringComparison.Ordinal)));
+            Assert.IsFalse(branches.Any(branch => string.Equals(branch.Name, "branch_1", System.StringComparison.Ordinal)));
```

Comment:
```suggestion
            var branches = branchClient.Search("main");
            var expectedBranch = branches.Single();
            Assert.AreEqual("main", expectedBranch.Name);
```

As the scenarios should only return 1 branch, calling single() makes the assert part a bit more readable (need only one assert that avoid possible confusion with the IsTrue/IsFalse sequence).

The same can likely be applied in most cases in [BranchClientTests.cs](https://github.com/ubisoft/NGitLab/pull/433/files#diff-7bdcc99ac007d231c7852e392552593d988c95a3af8a8593764d37299fb47720)